### PR TITLE
Added a GsonBuilder method to register a post deserialization annotat…

### PR DESCRIPTION
**Pull request by inder123**
…ion.

### CODE REVIEW

**Comment by inder123** [comment]
This addresses Issue https://github.com/google/gson/issues/269

File: gson/src/test/java/com/google/gson/functional/PostDeserializationAnnotationRegistrationTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2008 Google Inc.
```
Comment (cgruber) [inline gson/src/test/java/com/google/gson/functional/PostDeserializationAnnotationRegistrationTest.java]: This probably is a copy-pasta error.  2016?

Comment (inder123) [inline gson/src/test/java/com/google/gson/functional/PostDeserializationAnnotationRegistrationTest.java]: yes

**Comment by JakeWharton** [comment]
This is similar to what https://github.com/google/gson/pull/844 wanted.

You can currently write and register a delegating TypeAdapterFactory which checked for annotated methods post-deserialization without needing support in Gson itself. Is that enough?

**Comment by inder123** [comment]
@JakeWharton  I don't understand the delegating TypeAdapterFactory solution. Can you illustrate? Thanks.

**Comment by inder123** [comment]
This specific solution is good in one specific way: You can just map the javax.annotation.PostConstruct annotation and that will benefit users of frameworks such as JPA.

**Comment by inder123** [comment]
Can you suggest a better name for the GsonBuilder method, or even a different signature?
For example, should this method apply the annotation to All classes or match them with a regular expression?
Are there actions beyond post-deserialization where we could look for annotation? For example, pre/post serialization.

**Comment by swankjesse** [comment]
Here’s an example that does it standalone:
https://gist.github.com/swankjesse/20df26adaf639ed7fd160f145a0b661a

**Comment by swankjesse** [comment]
One other difference with this PR vs the gist above – this only applies to field bindings; that applies to any type.

**Comment by JakeWharton** [comment]
Nice @swankjesse. That's exactly what I was thinking.

**Comment by inder123** [comment]
Thanks @swankjesse for writing this up. I will try it out.
I think overall something like this needs to be part of Gson. We have talked about it for a very long time but without acting.

**Comment by inder123** [comment]
Closing this PR since the specified GIST is a workaround.

**Pull request closed without merging**